src/shared/ui/CustomSelect/CustomSelect.tsx

```diff
@@ -0,0 +1,76 @@
+import { FC, useState } from 'react';
+
+import cn from 'classnames';
+
+import { useToggle } from '~shared/model/useToggle';
+
+import css from './CustomSelect.module.scss';
+import { THandleOption } from './types/THandleOption';
+import { OptionButton } from './ui/OptionButton';
+
+type TCustomSelectProps = {
+  options: string[][];
+  startValue: string;
+  onChange?: (value: string) => void;
+};
+
+const CustomSelect: FC<TCustomSelectProps> = ({
+  options,
+  startValue,
+  onChange = () => {},
+}) => {
+  const [isModalOpen, toggle] = useToggle();
+  const [selectedOption, setSelectedOption] = useState(startValue);
+
+  const handleOption = ({ e, option, value }: THandleOption) => {
+    e.stopPropagation();
+
+    setSelectedOption(option);
+    onChange(value);
+    toggle();
+  };
+
+  return (
+    <div
+      tabIndex={-1}
+      role="button"
+      onClick={toggle}
+      className={css.containerSelect}
+      onKeyDown={() => {}}
+    >
+      <div className={cn(css.lableName)}>{selectedOption}</div>
+      <div
+        className={cn(css.menuSelect, {
+          [css.open]: isModalOpen,
+        })}
+      >
+        {options.map(option => (
```

Comment:
You should add an attribute key for the map to avoid errors from react

To improve the code, check for an empty options array (the component can handle when the options array is empty or undefined)   

Its optional